bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/philipstv/ConnectionManager.java

```diff
@@ -0,0 +1,129 @@
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    // Cannot use jetty in OH2.4 due to 9.4.11.v20180605 version with digest auth bug
+    // https://github.com/eclipse/jetty.project/issues/1555
+    private final CloseableHttpClient httpClient;
+
+    private final HttpHost httpHost;
+
+    public ConnectionManager(CloseableHttpClient httpClient, HttpHost httpHost) {
+        this.httpClient = httpClient;
+        this.httpHost = httpHost;
+        OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        OBJECT_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    public String doHttpsGet(String path) throws IOException {
+        String uri = httpHost.toURI() + path;
+        logger.debug(TARGET_URI_MSG, uri);
+        HttpGet httpGet = new HttpGet(uri);
+        String jsonContent = "";
+        try (CloseableHttpClient client = httpClient; //
+                CloseableHttpResponse response = client.execute(httpHost, httpGet)) {
+            validateResponse(response, uri);
+            jsonContent = getJsonFromResponse(response);
+        } catch (HttpHostConnectException e) {
+            logger.debug("HttpHostConnectException when getting {}", uri);
+        }
+        return jsonContent;
+    }
+
+    public String doHttpsPost(String path, String json) throws IOException {
+        String uri = httpHost.toURI() + path;
+        logger.debug(TARGET_URI_MSG, uri);
+        HttpPost httpPost = new HttpPost(uri);
+        httpPost.setHeader("Content-type", "application/json");
+        httpPost.setEntity(new StringEntity(json));
+        String jsonContent = "";
+        try (CloseableHttpClient client = httpClient; //
+                CloseableHttpResponse response = client.execute(httpHost, httpPost)) {
+            validateResponse(response, uri);
+            jsonContent = getJsonFromResponse(response);
+        } catch (HttpHostConnectException e) {
+            logger.debug("HttpHostConnectException when getting {}", uri);
+        }
+        return jsonContent;
+    }
+
+    private void validateResponse(CloseableHttpResponse response, String uri) throws HttpResponseException {
+        if (response == null) {
+            throw new HttpResponseException(0, String.format("The response for the request to %s was empty.", uri));
+        } else if (response.getStatusLine().getStatusCode() == 401) {
+            throw new HttpResponseException(401, "The given username/password combination is invalid.");
+        }
+    }
+
+    private String getJsonFromResponse(HttpResponse response) throws IOException {
+        String jsonContent = EntityUtils.toString(response.getEntity());
+        logger.debug("----------------------------------------");
+        logger.debug("{}", response.getStatusLine());
+        logger.debug("{}", jsonContent);
```

Comment:
Usually trace is used to request/response content. Debug for a higher level of statuses.